src/druidry/data_source.py

```diff
@@ -301,7 +296,11 @@ def get_metric(self, name):
         return next((m for m in self.metrics() if m.metric == name), None)
 
     def get_dimension(self, name):
-        return next((d for d in self.dimensions() if d.dimension == name), None)
+        dimension = getattr(self, name)
+        if isinstance(dimension, Dimension):
+            return dimension
+        else:
+            None
```

Comment:
Does it have to return None?

Reply:
i like it being explicit, the other option is making it raise an exception